handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -461,7 +464,7 @@ public final long sslCtxPointer() {
     //            get access to an OpenSslSessionContext after this method was called to prevent the user from
     //            producing a segfault.
     final void destroy() {
-        synchronized (ReferenceCountedOpenSslContext.class) {
+        if (DESTROYED_UPDATER.compareAndSet(this, 0, 1)) {
```

Comment:
@Scottmitch 2) sounds good to me. Let me ensure we do in all our methods in netty-tcnative.